### DISCUSSION TITLE
NextHead in _document makes the page load forever

### DISCUSSION THREAD

**Discussion by vladimirtomic-smg** [vercel/next.js#65082] [Help]
### Summary

We have a custom `Head` component which is wrapping `NextHead`. It looks like this:

```
import * as React from 'react';
import NextHead from 'next/head';

export interface HeadProps {
  title?: string | string[];
  children?: React.ReactNode;
}

export const Head: React.FC<HeadProps> = ({ title }) => {
  title = /* altering if needed */;

  return (
    <NextHead>
      <title key="mytitle">{`MySite ${title}`}</title>
    </NextHead>
  );
};
```

We need to have the same title on each page, so I just I tried putting it in `src/pages/_document.page.tsx`:

```
import { Head } from 'components/Head/Head';
import { Html, Main, NextScript } from 'next/document';

export default function Document(): JSX.Element {
  return (
    <Html translate="no">
      <Head />
      <body>
        <Main />
        <NextScript />
      </body>
    </Html>
  );
}
```

However, our app stopped working after this. The page was just showing a spinner, and nothing was happening. Nothing in the console, nothing suspicious in the Network tab (apart from a request staying in the "pending" state forever).

Then I reverted the `_document` file and made it look like this:

```
import { Html, Head, Main, NextScript } from 'next/document';

export default function Document(): JSX.Element {
  return (
    <Html translate="no">
      <Head />
      ...
```

...and I put our custom `Head` component in every individual page, for example:

```
// src/pages/about/index.page.tsx

import { Head } from 'components/Head/Head';

function About(props: AboutProps): JSX.Element {
   ...
   return (
      <OurPageLayout>
         <Head />
         ...
}
```

After this, everything started working normally, and every page had a proper title.

What could be the problem with putting this `NextHead` component in `_document`?

**Comment by icyJoseph** [comment]
Hi,

`next/head` is not meant to be, ever, used in `_document.

> The `<Head />` component used in `_document` is not the same as [next/head](https://nextjs.org/docs/pages/api-reference/components/head). The `<Head />` component used here should only be used for any `<head>` code that is common for all pages. For all other cases, such as `<title>` tags, we recommend using [next/head](https://nextjs.org/docs/pages/api-reference/components/head) in your pages or components

- https://nextjs.org/docs/pages/building-your-application/routing/custom-document#caveats

Interactivity and other stuff are not possible from `next/document`

Also:

> _document is only rendered on the server, so event handlers like onClick cannot be used in this file.
> `<Html>`, `<Head />`, `<Main />` and `<NextScript />` are required for the page to be properly rendered.

A better place to do this, would be `_app`.

**Reply by pearcircuitmike** [reply]
next/head does not work currently... has it been deprecated for the pages router?
